### ISSUE TITLE
Jib doesn't work with build process that aut

### ISSUE THREAD

**Issue by NickChapman** [GoogleContainerTools/jib#2286]
**Environment**: Java/Kotlin + SpringBoot + Docker + Debian (Sorry, not sure exactly what you're looking for here)

- *Jib version:* 2.0.0
- *Build tool:* Gradle 6.1.1
- *OS:* Debian

**Description of the issue**: Jib doesn't trigger build steps that auto-generate interfaces. Specifically, we're using swagger-codegen to generate interfaces for our API. In our `build.gradle` we have a task `generateApi` and then `compileSwaggerJava.dependsOn generateApi`. When running the Jib build, it complains that none of the `.class` files for the interfaces exist.

**Expected behavior**: Jib should detect all parts of the build OR the documentation should more clearly explain how to add `dependsOn` rules for Jib.

**Steps to reproduce**:

  1. Create a Gradle task that auto-generates some code, in this case you could probably do something as simple as move some required files outside of the directory and then just `cp` them in during the build. 
  2. Build a container with Jib.
  3. Notice that it hasn't run all of the steps that your Build normally does.

**`jib-gradle-plugin` Configuration**: 
```groovy
import io.swagger.codegen.config.CodegenConfigurator
import io.swagger.codegen.DefaultGenerator

buildscript {
    repositories {
        mavenCentral()
    }
    dependencies {
        classpath('io.swagger:swagger-codegen:2.2.3')
    }
}

plugins {
    id 'java'
    id 'org.jetbrains.kotlin.jvm' version '1.3.30'
    id 'org.springframework.boot' version "2.2.4.RELEASE"
    id 'io.spring.dependency-management' version "1.0.9.RELEASE"
    id 'org.jetbrains.kotlin.plugin.spring' version "1.3.40"
    id 'org.jetbrains.kotlin.kapt' version '1.3.41'
    id 'com.google.cloud.tools.jib' version '2.0.0'
}

jib.to.image = {Our GCR path}

def openApiGeneratedOutputDir = "$buildDir/generated/swagger"

configurations {
    ktlint
    toolsImplementation.extendsFrom implementation
}

sourceSets {
    swagger {
        java {
            srcDir "${openApiGeneratedOutputDir}/src/main/java"
        }
    }
    tools {}
}

version '1.0-SNAPSHOT'

sourceCompatibility = 1.8

repositories {
    google()
    mavenCentral()
}

ext {
    set('springCloudVersion', "Hoxton.SR1")
}

dependencies {
    implementation "com.auth0:java-jwt:3.8.0"
    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk8"
    implementation "org.jetbrains.kotlinx:kotlinx-coroutines-core:1.3.2"
    compile group: 'org.springframework.plugin', name: 'spring-plugin-core', version: '1.2.0.RELEASE'
    implementation "org.springframework.boot:spring-boot-starter"
    implementation "org.springframework.boot:spring-boot-starter-web"
    implementation "org.springframework.cloud:spring-cloud-gcp-starter"
    implementation "org.springframework.cloud:spring-cloud-gcp-starter-data-spanner"
    implementation "org.springframework.boot:spring-boot-starter-security"
    implementation "com.fasterxml.jackson.core:jackson-annotations"
    implementation "com.fasterxml.jackson.core:jackson-core"
    implementation "com.fasterxml.jackson.core:jackson-databind"
    implementation "com.fasterxml.jackson.module:jackson-module-kotlin"
    implementation "org.apache.commons:commons-csv:1.7"
    implementation "io.springfox:springfox-swagger2:2.9.2"
    runtime "com.fasterxml.jackson.datatype:jackson-datatype-jdk8"
    runtime "com.fasterxml.jackson.datatype:jackson-datatype-jsr310"
    implementation "io.springfox:springfox-swagger-ui:2.9.2"
    implementation "org.springframework.boot:spring-boot-devtools"
    toolsImplementation 'org.springframework.shell:spring-shell-starter:2.0.1.RELEASE'

    testImplementation "org.springframework.boot:spring-boot-starter-test"
    testImplementation "junit:junit:4.13-beta-3"
    testImplementation "org.mockito:mockito-core:2.25.1"
    testImplementation "com.nhaarman.mockitokotlin2:mockito-kotlin:2.1.0"
    testImplementation 'pl.pragmatists:JUnitParams:1.1.1'
    testImplementation "com.google.cloud:google-cloud-pubsub:1.42.0"

    swaggerImplementation "org.springframework.boot:spring-boot-starter-web"
    swaggerImplementation "io.swagger:swagger-annotations:1.5.22"

    kapt "org.springframework.boot:spring-boot-configuration-processor"
    ktlint "com.pinterest:ktlint:0.33.0"

    // Source Set Build Dependencies
    implementation sourceSets.swagger.output
    toolsImplementation sourceSets.main.output
    testImplementation sourceSets.tools.output
}

dependencyManagement {
    imports {
        mavenBom "org.springframework.cloud:spring-cloud-dependencies:${springCloudVersion}"
    }
}

tasks.withType(org.jetbrains.kotlin.gradle.tasks.KotlinCompile).all {
    kotlinOptions.jvmTarget = "1.8"
}

bootRun {
    systemProperty "spring.profiles.active", System.getProperty("spring.profiles.active")
}

task generateApi {
    def openApiDefinition = "$rootDir/api.yaml"
    inputs.file(openApiDefinition)
    outputs.dir(openApiGeneratedOutputDir)

    doLast {
        def config = new CodegenConfigurator()
        config.setInputSpec(openApiDefinition)
        config.setOutputDir(openApiGeneratedOutputDir)
        config.setLang('spring')
        config.setAdditionalProperties([
                modelPackage: 'com.example.model',
                apiPackage: 'com.example.api',
                dateLibrary: 'java8',
                interfaceOnly: true,
                useTags: true
        ])
        new DefaultGenerator().opts(config.toClientOptInput()).generate()
    }
}
compileSwaggerJava.dependsOn generateApi

clean.doFirst {
    delete(openApiGeneratedOutputDir)
}
```

**Comment by NickChapman** [comment]
@saturnism

**Comment by loosebazooka** [comment]
Jib will depend on the runtimeClasspath in 2.1.0 (unreleased) onwards. So I think this might work? Can you try it out? Or link to a full project (for which you have the buildfile above) so we can try it out?

**Comment by NickChapman** [comment]
@loosebazook added `tasks.jib.dependsOn sourceSets.main.runtimeClasspath` and it still fails.

**Comment by loosebazooka** [comment]
Can you provide us with a sample project to inspect this. If it's not in your runtime classpath, then we can't pick it up, but perhaps there's a way to correctly configure it beacuse `  implementation sourceSets.swagger.output` doesn't seem to be working?

**Comment by loosebazooka** [comment]
Can you confirm that `sourceSets.swagger.output` actually contains a value?

My current best guess is that the interactions between `sourceSet.swagger` and `sourceSet.main` aren't configured correctly.

It feels like you need to add something like (though I haven't tested this)
```
sourceSets {
  main {
    runtimeClasspath += sourceSets.swagger.output
  }
}
```

**Comment by chanseokoh** [comment]
Can't confirm, but according to the comments above, this is supposed to be fixed 2.1.0. Closing due to inactivity.
